test/dynamicKeys.tests.js

```diff
@@ -37,106 +37,97 @@ describe('Dynamic Secret', function () {
     return 'Bearer ' + jwt.sign({username: username}, keys[username], options);
   };
 
-  var tokenHandler = function (request, reply) {
-    reply(request.auth.credentials.username);
+  var tokenHandler = function (request, h) {
+    return request.auth.credentials.username;
   };
 
-  var getKey = function(req, token, callback){
+  var getKey = async function(req, token){
     getKey.lastToken = token;
     var data = jwt.decode(token);
-    Hoek.nextTick(function(){
-      callback(null, keys[data.username], info[data.username]);
-    })();
+
+    return {
+      key: keys[data.username],
+      extraInfo: info[data.username]
+    }
   };
 
-  var validateFunc = function(decoded, extraInfo, callback){
+  var validateFunc = function(decoded, extraInfo){
```

Comment:
Maybe we can declare this `async`, just to make sure the code is actually calling `await`.

Reply:
not sure the JS runtime will do that check (it will do the opposite for sure). but still will add it, maybe some linter will do that check